docs/getting-started/github-apps.rst

```diff
@@ -45,47 +44,117 @@ sync with the linked GitHub repository.
+With the Dockstore GitHub App installed, the synchronization is done automatically. When
 you add a new branch or release of a workflow on GitHub, Dockstore is notified,
 and Dockstore updates its copy of the workflow. For example, After publishing a new release
 of a workflow on GitHub, a new version of the workflow will be present in
 Dockstore shortly afterwards.
 
-For this to work, a ``/.dockstore.yml`` file is required in each GitHub repository you want
+For this to work, a ``/.dockstore.yml`` file is required in the root directory of each GitHub repository you want
 to associate with a workflow on Dockstore. A template for both workflows and services are shown below,
-along with explanations for each field. For every branch on GitHub that has one of these files, a corresponding entry
+along with explanations for each field. For every branch/release on GitHub that has one of these files, a corresponding entry
 will be made on Dockstore.
 
 Workflow YML File
 ++++++++++++++++++
+For a workflow, the ``/.dockstore.yml`` has the following general structure
 
 .. code:: yaml
 
    version: 1.2
    workflows:
-      - name: aligner
-        subclass: CWL
-        primaryDescriptorPath: /Dockstore.cwl
-        testParameterFiles:
-            - /test/dockstore.cwl.json
+      - name: <String>
+        subclass: <CWL | WDL | NFL | GXFORMAT2>
+        primaryDescriptorPath: <String>
+        testParameterFiles: <String Array>
 
 version
     The version of the .dockstore.yml schema. Currently at 1.2.
 workflows
     An array of workflows. Each element corresponds to a workflow on Dockstore.
-name
+name (optional)
     The optional workflow name that is used to uniquely identify workflows in repositories with multiple workflows.
+    **Each workflow listed must have a unique (or no) name.**
 subclass
-    The descriptor language used for the workflow. Supported values include CWL, WDL, and NFL.
+    The descriptor language used for the workflow. Supported values include CWL, WDL, NFL (Nextflow), and GXFORMAT2 (Galaxy).
 primaryDescriptorPath
-    The path to the primary descriptor file in the Git repository
-testParameterFiles
-    An array of paths to test parameter files in the Git repository.
+    The absolute path to the primary descriptor file in the Git repository
+testParameterFiles (optional)
+    An array of absolute paths to test parameter files in the Git repository.
+
+Ex. .dockstore.yml with a single workflow
+
+.. code:: yaml
+
+   version: 1.2
+   workflows:
+      - subclass: CWL
+        primaryDescriptorPath: /Dockstore.cwl
+        testParameterFiles:
+            - /test/dockstore.cwl.json
+
+The above ``.dockstore.yml`` is for a single workflow. Note that the name is not present since it is optional.
+
+Ex. .dockstore.yml with multiple workflows
+
+.. important:: The **name** field is an optional field used when a repository has multiple workflows in it that a user wants to register
```

Comment:
Sort of implies that you can't set it if there is only one workflow. Which you probably wouldn't, but you never know.

Reply:
What about something like:

> Though the **name** field is an optional field when a ``.dockstore.yml`` has one workflow in it,  it must be used when a ``.dockstore.yml`` has multiple workflows in it. Each entry within a ``.dockstore.yml`` file corresponds to a unique entry on Dockstore.